### PR TITLE
f-checkout@0.44.0 When not logged in, retrieve serviceType from basket endpoint

### PR DESCRIPTION
**Pull request by pkedziora**
When the user is not logged in, retrieve serviceType from global basket orchestrator
This is needed because main Checkout GET endpoint requires authentication now

## UI Review Checks

No UI changes

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 1]
```diff
@@ -246,6 +257,10 @@ export default {
          */
         async initialise () {
             this.setAuthToken(this.authToken);
+            if (!this.isLoggedIn) {
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue:261]: Ultra nit: add an empty line before this one, please.

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 2]
```diff
@@ -327,11 +330,12 @@ describe('Checkout', () => {
                     localVue,
                     propsData
                 });
+                await flushPromises();
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:333]: Just curious, but why did we need to add this `flushPromises` in these tests? 🙂

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 3]
```diff
@@ -341,9 +345,27 @@ describe('Checkout', () => {
...
 
                 expect(loadAvailableFulfilmentSpy).toHaveBeenCalled();
             });
+
+            describe('if isLoggedIn set to `false`', () => {
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:353]: We should have the opposite case tested as well. If it's set to `true`, we should verify there's no call made.

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 4]
```diff
@@ -948,6 +970,44 @@ describe('Checkout', () => {
             });
         });
 
+        describe('loadBasket ::', () => {
+            describe('when `getBasket` request fails', () => {
+                let wrapper;
+
+                beforeEach(async () => {
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js]: We don't really need the `beforeEach` in here and the case below, given there's only one test. I'd personally remove it, but up to you.

File: packages/components/organisms/f-checkout/src/store/checkout.module.js [hunk 5]
```diff
@@ -139,6 +139,37 @@ export default {
...
+            tenant,
+            language,
+            timeout
+        }) => {
+            // TODO: deal with exceptions.
+            const config = {
+                method: 'get',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Accept-Tenant': tenant,
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/store/checkout.module.js:159]: We had to remove this from other the Checkout API calls. Do we need it for this API?

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js (hunk 2 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:333]: The test framework seems to return from mount to the test after first await. I now added another call so the test doesn't wait for this call to complete unless the flushPromises is added

File: packages/components/organisms/f-checkout/src/store/checkout.module.js (hunk 5 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/store/checkout.module.js:159]: Yep. The basket orchestrator calls in production have it

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js (hunk 4 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js]: Fixed

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js (hunk 3 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:353]: Fixed

File: packages/components/organisms/f-checkout/src/components/Checkout.vue (hunk 1 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue:261]: Fixed

File: packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js [hunk 6]
```diff
@@ -195,6 +198,33 @@ describe('CheckoutModule', () => {
...
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Accept-Tenant': payload.tenant,
+                        'Accept-Language': payload.language
+                    },
+                    timeout: payload.timeout
+                };
+
+                axios.get = jest.fn(() => Promise.resolve({ data: basketDelivery }));
+            });
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js]: Same as before. We could remove this `beforeEach` in this case. Sorry, didn't catch it in my original review.

Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js]: Fixed

**Review by SachaZvetelman — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/store/checkout.module.js [hunk 7]
```diff
@@ -214,6 +245,10 @@ export default {
                 ...state.customer,
                 ...customer
             };
+        },
+
+        UPDATE_BASKET_DETAILS (state, { serviceType }) {
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-checkout/src/store/checkout.module.js]: Shouldn't these follow the same syntax setup as the other mutations? i.e the `:` & `=>` are missing here and in a few other places. I think it still fine without tho so happy to leave as is for now, might be one for making consistent later.

File: packages/components/organisms/f-checkout/src/store/checkout.module.js [hunk 8]
```diff
@@ -139,6 +139,37 @@ export default {
...
+                    'Content-Type': 'application/json',
+                    'Accept-Tenant': tenant,
+                    'Accept-Language': language
+                },
+                timeout
+            };
+
+            const { data } = await axios.get(url, config);
+            const basketDetails = {
+                serviceType: data.ServiceType.toLowerCase()
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-checkout/src/store/checkout.module.js:167]: Do you need to check that data exists before trying to access it? Will it always be available? If not you might get an error at `ServiceType`/

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js [hunk 9]
```diff
@@ -948,6 +987,36 @@ describe('Checkout', () => {
...
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData
+                });
+
+                it('should emit success event', async () => {
+                    expect(wrapper.emitted(EventNames.CheckoutBasketGetSuccess).length).toBe(1);
+                    expect(wrapper.emitted(EventNames.CheckoutBasketGetFailure)).toBeUndefined();
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:1021]: Probably worth having this a separate `IT`:

`it('should not emit a 'CheckoutBasketGetFailure' when basket is successful')`

Comment (kevinrodrigues) [reply packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:1021]: In fact i dont think you need this one as the one above looks to cover the failure. So this test really only needs the expect on the line above. Then the test above only needs the failure one.

File: packages/components/organisms/f-checkout/src/store/checkout.module.js (hunk 7 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/store/checkout.module.js]: Fixed

File: packages/components/organisms/f-checkout/src/store/checkout.module.js (hunk 8 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/store/checkout.module.js:167]: It should always be available

File: packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js (hunk 9 above)
Comment (pkedziora) [reply packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js:1021]: Fixed

**Review by kevinrodrigues — APPROVED** [review]

**Review by SachaZvetelman — APPROVED** [review]

**Pull request merged**